### PR TITLE
Implement /prices API

### PR DESCRIPTION
**Pull request by jniles**
This PR implements the price list api under the `/prices` endpoint.  More specifically,
- `GET /prices` returns a list of price lists
- `GET /prices/:uuid` returns the details of a single price list + items
- `POST /prices` creates a new price list (and optionally, price list items)
- `PUT /prices/:uuid` updates an existing price list + items
- `DELETE /prices/:uuid` will delete a price list + items.

Included in this PR are tests for all routes.  Some corner cases tested:
1. Creating a price list with price list items
2. Creating a price list without price list items
3. Deleting a price list with price list items
4. If a price list item is invalid, the request fails gracefully and cleans up the created price list.

I recommend the #10 be merged before this PR is merged.  The tools found there will help clean up these integration tests immensely.

### CODE REVIEW

File: server/controllers/finance/priceList.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+};
+
+/**
+ * Convert an array of price list item objects into an array of arrays for
+ * easy database insertion
+ *
+ * @param {String} priceListUuid
+ * @param {Array} items
+ *
+ * @example
```
Comment (sfount) [inline server/controllers/finance/priceList.js:122]: Mad comments :ok_hand:

File: server/controllers/finance/priceList.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    // wrap the queries in a DB transaction.
+    if (!isEmptyObject(data)) {
+      trans.addQuery(priceListSql, [ data, req.params.uuid ]);
+    }
+
+    // only trigger price list item updates if the items have been sent back to
+    // the server
+    if (items) {
+      trans.addQuery(priceListCreateItemSql, [ items ]);
+      trans.addQuery(priceListDeleteItemSql, [ req.params.uuid ]);
```
Comment (sfount) [inline server/controllers/finance/priceList.js:272]: This has been mentioned but addQuery returns the base object allowing you to chain the function. 

``` js
trans
   .addQuery(sql, [])
   .addQuery(sql, []);
```

Comment (jniles) [inline server/controllers/finance/priceList.js:272]: :+1:

File: server/controllers/finance/priceList.js [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+  .done();
+};
+
+
+/**
+ * Delete a price list (and associated items) in the database.
+ *
+ * DELETE /prices/:uuid
+ */
+exports.delete = function del(req, res, next) {
```
Comment (sfount) [inline server/controllers/finance/priceList.js:295]: We should standardise module export style for future modules. I don't have too strong an opinion either way. 

As I understand the two options are:
**Declare export references at the top**

``` js
exports.create = create;
exports.delete = delete;
```

**Declare exports by assigning methods directly**

``` js
exports.create = function create()...
exports.delete = function delete()...
```

@IMA-WorldHealth/local-contributors

Comment (jniles) [inline server/controllers/finance/priceList.js:295]: I have no preference either way.   Which one works better with JSDoc @sfount @mbayopanda ?
